doc/rationale.md

```diff
@@ -0,0 +1,31 @@
+# Ignition rationale
+
+Ignition is a distribution-agnostic provisioning utility. When a machine boots for the first time, Ignition will manipulate it in various ways to set up the machine for operations.
+
+There are several principles that have guided Ignition's design. This document aims to explain these principles and how they impact Ignition's design.
```

Comment:
This document will explain the principles used, and the impact they have had on Ignition's design.